cli/dts/lib.deno.unstable.d.ts

```diff
@@ -1214,4 +1214,32 @@ declare namespace Deno {
    * The pid of the current process's parent.
    */
   export const ppid: number;
+
+  /** **UNSTABLE**: New API, yet to be vetted.
+   * A custom HTTPClient for a fetch.
+   * 
+   * ```ts
+   * const client = new Deno.HTTPClient({ caFile: "./ca.pem" });
+   * const req = await fetch("https://myserver.com", { client });
+   */
+  export class HTTPClient {
+    rid: number;
+    constructor(init: HTTPClientInit);
```

Comment:
Wouldn't the more consistent API be `Deno.createHttpClient()` as opposed to a constructor? E.g. `Deno.run()`. It's not the best thing to allocate resources in constructors.
```ts
Deno.createHttpClient(options: Deno.CreateHttpClientOptions);
```

Reply:
I guess. I personally prefer `new Deno.HttpClient()` - adding a factory function just because seems unnecessary. We sort of have precedent with `Deno.Buffer`, although that does not allocate resources in the resource table.

Reply:
I just think it says something that, if it happened to be rely on an async op, you'd have to use a different language construct to represent it as opposed to just adding `async/await` here and there.

You're only supposed to do basic data stationing in constructors that doesn't mess with global state.

Reply:
But it's more API surface area. Seems odd to me.

Reply:
We can exclude the constructor from the types https://github.com/denoland/deno/pull/6918#discussion_r465344513, and the surface area is the same unless we're counting symbols. Constructors aren't meant for this stuff or you'd be able to make them async, for example.